src/Symfony/Bundle/FrameworkBundle/Resources/config/messenger.xml

```diff
@@ -52,7 +52,7 @@
 
         <!-- Logging & Debug -->
         <service id="messenger.middleware.debug.logging" class="Symfony\Component\Messenger\Debug\LoggingMiddleware">
```

Comment:
Another workaround could be creating from scratch this service definition in `MessengerPass`, only if debug and logger are enabled/exists, wdyt?

Reply:
We usually try as much as possible to declare all services in xml files so everything is in one place, favoring discovery/inspections and allowing to jump from classes to definition with the PhpStorm Symfony plugin.